extensions/amp-tiktok/validator-amp-tiktok.protoascii

```diff
@@ -32,7 +32,7 @@ tags: {  # <amp-tiktok>
   attr_lists: "extended-amp-global"
   attrs: {
     name: "data-src";
-    value_regex: "\\d";
+    value_regex: "(.+/)?\\d+/?";
```

Comment:
Why `(.+/)?`?

Reply:
Because `data-src` can either look like:

* `6943753342808034566`
* `https://www.tiktok.com/@scout2015/video/6943753342808034566/`

See the regex in the JS for how the ID gets parsed out of the string:

https://github.com/ampproject/amphtml/blob/f0a90fae17c7224f83f40dddb6583aef4e7d96d1/extensions/amp-tiktok/0.1/amp-tiktok.js#L168-L170

Reply:
That doesn't mean we should just validate it against what the J/S code does. Right now it's a single digit for validation. Why not require it be just the digits of the ID?

Reply:
According to `validator-amp-tiktok.html` the intention seemed to be to allow a TikTok URL as well as a TikTok ID:

https://github.com/ampproject/amphtml/blob/f0a90fae17c7224f83f40dddb6583aef4e7d96d1/extensions/amp-tiktok/0.1/test/validator-amp-tiktok.html#L37-L45

@rebeccanthomas could you provide more insights?

Reply:
I agree that limiting it to an ID makes more sense, but I don't have the background on the component design.

Reply:
In this case we are allowing both because of the way the TikTok api provides a placeholder image. In order to request the placeholder image we need to use an oEmbed request URL, therefore in order to allow the TikTok's to display a placeholder we chose to allow both the videoId and the full URL.

My original thought for validation was to verify that there is at least 1 digit in the src, and then any further processing would occur inside of the buildCallback of the extension.

Reply:
In that case, what about making the pattern either allow just an ID or specifically match a TikTok video URL?

```suggestion
    value_regex: "\\d+|https://www\.tiktok\.com/@[a-zA-Z0-9_.]+/video/\\d+/?";
```

Reply:
My only concern would be that we would need to keep an eye on the TikTok API to make sure that the URL doesn't change shape but perhaps that could be solved by having a more general shape for the regex such as:  `\\d+|https://www\.tiktok\.com/[a-zA-Z0-9_.]+\\d+/?` Which would allow for a bit more flexibility?

Reply:
Humm. That wouldn't match `https://www.tiktok.com/@scout2015/video/6943753342808034566/`

Reply:
Two things, the validator...

1. isn't meant to keep developers from doing something they shouldn't do. It's meant to preserve the security and privacy of AMP.
2. where possible provide developers with guidance.

Going to say the thing everyone hates saying, we had one problem then we introduced regex now we have two problems. I'd rather avoid adding more regex, especially if it's complicated.

IMO, if the current ID is a series of digits by their API then we should validate it as such (what we have now except a lower and upper limit).

If we're explicitly allowing URLs in this then we should think carefully about it. As Weston pointed out even a rudimentary pattern might not fit. Why can we not ask tik-tok component users to just use the ID? If we want to use the URL pattern then we should embrace the current URL pattern knowing that at some point it may change

Reply:
+1 And we know that nobody has been able to use URLs successfully until now anyway because they have been invalid. So deciding to go with digits alone for now will make the first example valid. Since the component is brand new, users won't be missing the URL usage.

Reply:
@rebeccanthomas Actually, in looking at the component JS, it seems that the component behaves best when a URL is provided as then it is able to provide a placeholder. Is that right? Is supplying video ID digits as `src` the worst usage of the component?

Reply:
Thats right Weston, the most polished behavior of the component happens when they provide the URL. The reason we chose to support just having the ID was because that is technically the minimum information needed to display a TikTok however it might be doable to ask users to simply provide the full URL

@alanorozco - Do you think removing the ID only src option for TikTok would work design wise?

Reply:
Yes, the ID-only usage is not preferred since that has less functionality.

A regex like `^(.*)(\d+)(.*)$` would be compatible with both APIs and likely be loose enough for the future. Could we use that?

Reply:
Any reason we don't include a protocol (`https`) with the regex? I know this isn't pre-rendered but it would be nice to at least require it.

Reply:
What about HTTPS with the `www.tiktok.com` domain?

```suggestion
    value_regex: "(https://www\.tiktok\.com/.*)?\\d+";
```

Reply:
I'm game with either:

```suggestion
    value_regex: "(https://.+)?\\d+.*";
```
```suggestion
    value_regex: "(https://www\.tiktok\.com/.*)?\\d+.*";
```